modules/globebrowsing/util/globelabelscomponent.cpp

```diff
@@ -0,0 +1,842 @@
+                }
+
+                if (dictionary.hasKey(LabelsDistanceEPSInfo.identifier)) {
+                    float dist = static_cast<float>(
+                        dictionary.value<double>(LabelsDistanceEPSInfo.identifier)
+                    );
+                    _labelsDistaneEPS.set(dist);
+                }
+
+                if (dictionary.hasKey(LabelAlignmentOptionInfo.identifier)) {
+                    std::string labelAlignment = 
+                        dictionary.value<std::string>(LabelAlignmentOptionInfo.identifier);
+                    if (labelAlignment == "Horizontally") {
+                        _labelAlignmentOption = 0;
+                    }
+                    else {
+                        _labelAlignmentOption = 1;
+                    }
+                }
+
+                _font = font;
+                initializeFonts();
+            }
+        }
+    }
+}
+
+bool GlobeLabelsComponent::initializeGL() {
+    return true;
+}
+
+void GlobeLabelsComponent::initializeFonts() {
+    if (_font == nullptr) {
+        _font = openspace::global::fontManager.font(
+            "Mono",
+            static_cast<float>(_labelsFontSize),
+            ghoul::fontrendering::FontManager::Outline::Yes,
+            ghoul::fontrendering::FontManager::LoadGlyphs::No
+        );
+    }
+}
+
+
+bool GlobeLabelsComponent::deinitialize() {
+    return true;
+}
+
+bool GlobeLabelsComponent::isReady() const {
+    return true;
+}
+
+void GlobeLabelsComponent::update() {
+        
+}
+
+bool GlobeLabelsComponent::loadLabelsData(const std::string& file) {
+    bool success = true;
+    if (_labelsDataPresent) {
+        std::string cachedFile = FileSys.cacheManager()->cachedFilename(
+            ghoul::filesystem::File(file),
+            "GlobeLabelsComponent|" + identifier(),
+            ghoul::filesystem::CacheManager::Persistent::Yes
+        );
+
+        bool hasCachedFile = FileSys.fileExists(cachedFile);
+        if (hasCachedFile) {
+            LINFO(fmt::format(
+                "Cached file '{}' used for labels file '{}'",
+                cachedFile,
+                file
+            ));
+
+            success = loadCachedFile(cachedFile);
+            if (success) {
+                return true;
+            }
+            else {
+                FileSys.cacheManager()->removeCacheFile(file);
+                // Intentional fall-through to the 'else' to generate the cache
+                // file for the next run
+            }
+        }
+        else {
+            LINFO(fmt::format("Cache for labels file '{}' not found", file));
+        }
+        LINFO(fmt::format("Loading labels file '{}'", file));
+
+        success = readLabelsFile(file);
+        if (!success) {
+            return false;
+        }
+
+        success &= saveCachedFile(cachedFile);
+    }
+    return success;
+}
+
+bool GlobeLabelsComponent::readLabelsFile(const std::string& file) {
```

Comment:
Would be great to add some documentation about the expected columns of the csv file, if people want to generate their own

Reply:
I will create a doc in the week about this.